packages/dom/src/dom/clean-node-list.js

```diff
@@ -11,27 +11,44 @@ import remove from './remove';
 import unwrap from './unwrap';
 import { isPhrasingContent } from '../phrasing-content';
 import insertAfter from './insert-after';
+import isElement from './is-element';
+
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @typedef SchemaItem
+ * @property {string[]} [attributes] Attributes.
+ * @property {(string | RegExp)[]} [classes] Classnames or RegExp to test against.
+ * @property {'*' | { [tag: string]: SchemaItem }} [children] Child schemas.
+ * @property {unknown[]} [require] ???
```

Comment:
Judging by how this is treated later, should it be an array of strings, or perhaps an array of elements? Although the latter sounds a bit limiting. Maybe either could be supported?

See https://github.com/WordPress/gutenberg/blob/6156092019cee0ffe1a9c0f9a63331ff90b30c66/packages/dom/src/dom/clean-node-list.js#L122